src/api/java/com/minecolonies/api/research/LocalResearchTree.java

```diff
@@ -0,0 +1,136 @@
+package com.minecolonies.api.research;
+
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.research.interfaces.ILocalResearch;
+import com.minecolonies.api.research.util.ResearchState;
+import com.minecolonies.api.util.NBTUtils;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+
+import static com.minecolonies.api.research.util.ResearchConstants.*;
+
+/**
+ * The class which contains all research.
+ */
+public class LocalResearchTree
```

Comment:
Again this should be abstracted away into an interface if need be and then this implementation should not be in the api.